### PR TITLE
[raz] Fix rename and cases of % in path for RAZ ABFS

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Fixing creation of directories and other directory operation having % in name/path for ABFS. For this, we have to double-encode such path. This is required because we do the calls to ABFS in single encoding and the underlying HTTPClient takes care of unquoting it once already. 
- For the RAZ side, we make sure that we unquote such paths once before making the RAZ request to get the SAS token.

<br>

- We are also making sure that we send correct path in-case of actual whitespaces vs actual `%20` present in directory paths. 
- This means that we only **double encode** actual %20 paths and not the paths which contain a whitespace (because encoding a whitespace gives `%20`).

<br>

- Fix rename in Py2 setup for non-ascii and % in path scenarios for ABFS.
- In case of Py2 setup, we are UTF-8 converting the unicode to str for correct URL quoting.

## How was this patch tested?

- Manually E2E tested in RAZ enabled cluster.
- Added new unit tests for different scenarios and updated some existing tests.

### CODE REVIEW

File: desktop/core/src/desktop/lib/rest/raz_http_client.py [hunk 1]
```diff
@@ -47,6 +49,12 @@ def execute(self, http_method, path, params=None, data=None, headers=None, allow
     sig=pJL%2FWyed41tptiwBM5ymYre4qF8wzrO05tS5MCjkutc%3D&st=2015-01-02T01%3A40%3A51Z&se=2015-01-02T02%3A00%3A51Z&sp=r
     """
 
+    if '%' in path:
```
Comment (bjornalm) [inline desktop/core/src/desktop/lib/rest/raz_http_client.py:52]: Why this condition? If you need the path to be encoded in your url then why not always encode it?

Comment (Harshg999) [reply desktop/core/src/desktop/lib/rest/raz_http_client.py:52]: Good catch! I tested in live cluster and there's no harm in doing it for all path. I've done the necessary changes but I've found a small bug (not related to this) in which we are not able to go in the `%20` directories (although few other operations are working fine for them). So, I'm marking this PR as draft for now!

Comment (bjornalm) [reply desktop/core/src/desktop/lib/rest/raz_http_client.py]: Sounds good

**Comment by Harshg999** [comment]
Tested the changes on a live cluster. I've also added unit tests for different scenarios. 
The PR looks good and ready for review now!

**Review by ranade1 — APPROVED** [review]
looks good to me.

**Review by wing2fly — APPROVED** [review]

File: desktop/core/src/desktop/lib/rest/raz_http_client.py (hunk 1 above)
Comment (bjornalm) [reply desktop/core/src/desktop/lib/rest/raz_http_client.py]: It looks that the condition is still present in the merged code. Did you forget or was there a good reason for adding it? In general, I strongly feel we should should avoid conditional encoding and stick with a consistent strategy: I.e. if the string is passed as a url parameter and **might** contain characters that need encoding, then we should always encode it. Conditionally encoding these strings will make future debugging much harder...

**Pull request merged**